examples/aes-ccm/src/ccm_mode.rs

```diff
@@ -0,0 +1,197 @@
+use std::convert::TryInto;
+use hacspec_lib::*;
+use hacspec_aes::*;
+
+fn format_func(a: &ByteSeq, n: &ByteSeq, p: &ByteSeq, t: u8, alen: u64, nlen: u8, plen: u64) -> ByteSeq {
+    let mut r = 0;
+    let mut tmp = 0;
+
+    if alen < (1<<16 - 1<<8) {
+        tmp = 2;
+    } else if alen < (1<<32) {
+        tmp = 6;
+    } else {
+        tmp = 10;
+    }
+
+    r += ((tmp+alen+15)/16)+((plen+15)/16); // ceiling operation used
+    let mut b = ByteSeq::new((16*(r+1)).try_into().unwrap());
+
+    // creation of b(0)
+    let qlen: u8 = 15 - nlen;
+    let mut flags: u8 = 0;
+
+    if alen > 0 {
+        flags = 0x40;
+    }
+
+    flags = flags | (((t-2)/2) << 3);
+    flags = flags | (qlen-1);
+    let f = ByteSeq::from_public_slice(&[flags]);
```

Comment:
This creates an array and isn't allowed. Unfortunately the standard library still offers these functions (for testing). Also see #136 for why this doesn't work well yet. The best thing to do here right now is to use pre-defined `array!`s instead of `ByteSeq` unless you really need a `ByteSeq`.

Reply:
I'm not sure how I can do this without a `ByteSeq`. I have to insert the flags as one byte of `b`. I think the only way to do that is through `set_exact_chunk` right? And it requires a `ByteSeq f` to be passed as the parameter to set it in the larger `ByteSeq b`.

Reply:
It looks like you want to modify a single byte in `b`. `ByteSeq` (should) implement `IndexMut`. So you can just modify it directly, e.g. `b[0] = flag`.

Reply:
That isn't working. It gives me an error saying 
```
 ^^^^ expected struct `U8`, found `u8`
```
Changing `flags` to type U8 also won't work since I am modifying it with other `u8` variables...

Reply:
You can modify `U8`s. This works fine when converting all necessary variables to `U8`. You can also compute on `u8` and then convert it only when adding it into `b` if it doesn't need to be a `U8`, e.g. `b[0] = U8(flags)`;

Reply:
Ohh, that works. Thanks!